### ISSUE TITLE
Shipping Sly with support for the Community Spec

### ISSUE THREAD

**Issue by enzuru** [joaotavora/sly#579]
Hey @fonol, you may not remember me, but I responded to your post on Reddit a long time ago about how the GCL was a Common Lisp spec with a friendlier license. 

Since then you've done amazing things and for the first time ever perhaps, we have a really good-looking and friendly Common Lisp "community spec": https://cl-community-spec.github.io/

I would like to now open a ticket with @joaotavora so that we can discuss perhaps shipping Sly with community spec support, and not merely HyperSpec support. Obviously this "community spec" has a license that is more compatible with the GNU operating system.

The code most likely should be easy enough, seems to be `https://cl-community-spec.github.io/pages/${symbol}.html` in the vast majority of cases.

But perhaps there are other issues to consider?

**Comment by fonol** [comment]
Hi, 
glad that you find the project useful. I don't use Emacs, but I guess it could be nice to have support for the spec there. 
I would consider though that at the moment, the HyperSpec is certainly more complete and error-free. The CommunitySpec contains missing links, probably also some wrong links (e.g. stuff like "nil" sometimes linking on the wrong of the three pages about `nil`), wrong formatting, and also completely misses any images/figures (I think there are some in the HyperSpec?).

For the naming, it is mostly as you said that the symbol name is just the name of the HTML file. However, there are symbols not allowed in the URL which have to be escaped with their ASCII code, e.g. [002abreak_002don_002dsignals_002a.html](https://cl-community-spec.github.io/pages/002abreak_002don_002dsignals_002a.html). For convenience, there is a mapping of all the terms and pages here: https://github.com/fonol/cl-community-spec/blob/main/searchable_terms.json.

**Comment by charJe** [comment]
Even just opening the appropriate community spec page in the operating system default browser would be very nice.

**Comment by jgarte** [comment]
This would be a very nice feature to have in sly.

**Comment by jgarte** [comment]
@joaotavora 

Do you think that this should be a sly contrib package instead and we should close this issue?

**Comment by joaotavora** [comment]
> Do you think that this should be a sly contrib package instead and we should close this issue?

No, and sorry for not replying earlier (there wasn't a question until now, though).

I think you should attempt to do as many changes to `lib/hyperspec.el` as needed or possibly create a `lib/cl-community-spec.el` file so that the user can use this spec instead of the Hyperspec.  Just make sure that the user can easily toggle between them (`sly-documentation-function` looks like it could do the job) and the interface to using them (say `C-c C-d C-h`) is the same.

`browse-url` should be used once an URL is computed.  This ensures Emacs does the right thing (shows it in the in-editor browser if available and preferred, otherwise picks another browser).

BTW, this is unrelated, but Emacs CL emulation package (`cl-lib.el`) is "under attack" in the Emacs community (they're real fans of Greenpuns's 10th there) and one of the things they attack it with is the lack of good documentation.  Having Lispworks' Hyperspec as a non-free resource or something that can be worked from didn't help.  So I'm happy you're refreshing my memory on this great work.

**Comment by jcguu95** [comment]
@enzuru The hardest part of this project is to create a hash-table that maps the symbols to url. It is done in `hyperspec.el`.

``` elisp
(mapc (lambda (entry)
	(common-lisp-hyperspec--insert (car entry) (cadr entry)))
      if common-lisp-hyperspec-symbol-table
	  (common-lisp-hyperspec--parse-map-file
	   common-lisp-hyperspec-symbol-table)
	'("&allow-other-keys" "03_da.htm")
	  ("&aux" "03_da.htm")
	  ("&body" "03_dd.htm")
[elided..]
```

A possible and good enough workaround is to send search query to `https://cl-community-spec.github.io/` and open up the searched results. Something like `https://cl-community-spec.github.io/MAH-OWN-INVENTED-API-for--search/unwind-protect`. Either way, we need some insights from @fonol .
